pydicom/dataset.py

```diff
@@ -1379,13 +1374,13 @@ def setdefault(
                 try:
                     vr = dictionary_VR(tag)
                 except KeyError:
-                    if config.enforce_valid_values:
+                    if (config.settings.writing_validation_mode ==
+                            config.RAISE):
```

Comment:
As soon as the user sets `read_validation_mode` once, yes, `enforce_valid_values` will be ignored. Before `enforce_valid_values` was always used if it was set to `True`. In both cases it is a problem if the user mixes both properties. Without a possibility to implement `__setattr__` at module level, I see no way to prevent this.

Reply:
Ah, wait - this is about `writing_validation_mode` - no there is no fallback, as there was no check for writing before at all. Do we need a fallback to also use `enforce_valid_values` for that case?